pkg/cloudprovider/provider/fake/provider.go

```diff
@@ -91,3 +91,7 @@ func (p *provider) MigrateUID(machine *v1alpha1.Machine, new types.UID) error {
 func (p *provider) MachineMetricsLabels(machine *v1alpha1.Machine) (map[string]string, error) {
 	return map[string]string{}, nil
 }
+
+func (p *provider) SetMetricsForMachines(_ v1alpha1.MachineList) error {
+	return fmt.Errorf("Not implemented")
```

Comment:
Why return for some implementations a `Not implemented` errors and for some simply nil?
https://github.com/kubermatic/machine-controller/pull/444/files#diff-42c33a2522ded90fccbad92e902b319cR358

I would prefer going the `nil` way as metrics are optional and should not cause any errors